contents/docs/contribute/developing-locally.mdx

```diff
@@ -9,231 +9,268 @@ showTitle: true
+This installs both the Postgres server and its tools. DO NOT start the server after running this.
 
-```bash
-python3 -m venv env
-```
+On Linux you often have separate packages: `postgres` for the tools, `postgres-server` for the server, and `libpostgres-dev` for the `psycopg2` dependencies. Consult your distro's list for an up to date list of pacakages.
 
-3. Activate the virtual environment:
 
-```bash
-source env/bin/activate
-```
+### Step 2. NodeJS Frontend
 
-4. Install requirements with pip
+1. Install nvm by following the [instructions here](https://github.com/nvm-sh/nvm)
 
-```bash
-pip install -r requirements.txt
-```
+2. Install the latest minor release of NodeJS 14 with `nvm install 14`.
 
-If you have problems with this step (TLS/SSL error), then run
+3. Install yarn version 1.22 with `npm install -g yarn@1`
 
-```bash
-~ brew update && brew upgrade
-```
+4. Install npm packages by running `yarn`
 
-followed by
+5. Run `yarn start`, which runs in parallel 1) our [esbuild](https://esbuild.github.io/)-powered dev server, 2) [kea-typegen](https://github.com/keajs/kea-typegen), which generates `logicType.ts` files for each `logic.ts` in the codebase.
 
-```bash
-python3 -m pip install --upgrade pip
-```
+6. The first time you run typegen, it may get stuck in a loop. If so, cancel the process (`ctrl+c`), discard all changes in the working directory (`git reset --hard`), and run `yarn start` again. You may need to discard all changes once more when the second round of type generation completes.
 
-then retry the requirements.txt install.
+### Step 3. NodeJS Plugin Server
 
-> **Friendly tip:** If you're running into errors with this step on a Mac with an Apple Silicon chip, you may need to set some environment variables before running the install command, like so:
-    
-```bash
-CFLAGS="-I /opt/homebrew/opt/openssl/include" LDFLAGS="-L /opt/homebrew/opt/openssl/lib" GRPC_PYTHON_BUILD_SYSTEM_OPENSSL=1 GRPC_PYTHON_BUILD_SYSTEM_ZLIB=1 pip install -r requirements.txt
-```
-    
-5. If you want to fully test all our features, you'll need to install a few dependencies for SAML to run properly. If you're on macOS run the command below, otherwise check out the official [xmlsec repo](https://github.com/mehcode/python-xmlsec) for more details.
+1. Assuming NodeJS is installed, run `cd plugin-server && yarn` to install all required packages.
+
+### Step 4. Python Backend
+
+1. Install pyenv by following the [instructions here](https://github.com/pyenv/pyenv).
+
+2. Install the latest path release of Python 3.9 with `pyenv install 3.9` (and then selecting the precise version from the list presented). For example:
 
 ```
-brew install libxml2 libxmlsec1 pkg-config
-pip install python3-saml==1.12.0
+pyenv install 3.9 # gives a list of versions to install
+pyenv install 3.9.9
+pyenv global 3.9.9
 ```
 
-6. Install dev requirements:
+3. Create the virtual environment in current directory called 'env':
```

Comment:
If we're using pyenv then manually creating venvs is unneeded. You can do this instead:

- `pyenv virtualenv 3.9.9 posthog`
- Use some way to automatically activate that when you cd into the dir - I use https://github.com/eknkc/fish-autoenv